cmd/homi/setup/cmd.go

```diff
@@ -701,16 +704,15 @@ func Gen(ctx *cli.Context) error {
 		nodeAddrs []common.Address
 	)
 
-	if len(ctx.String(mnemonic.Name)) == 0 {
-		privKeys, nodeKeys, nodeAddrs = istcommon.GenerateKeys(cnNum)
-	} else {
-		mnemonic := ctx.String(mnemonic.Name)
+	if keydir := ctx.String(cnNodeKeyDirFlag.Name); len(keydir) > 0 {
+		privKeys, nodeKeys, nodeAddrs = istcommon.LoadNodekey(keydir)
```

Comment:
Tested with wrong keypath and it returned panic caused by line 732 `privKeys[:numValidators]`. Do you think we need to handle this?

Reply:
Yep, PN and EN flags are checked but CN flag is not. Added a8f9a3d

Reply:
Maybe `cnNum` is more precise here? lines 726 and 728 use `cnNum`.

Reply:
fixed 7b23fff